src/services/database_service.py

```
@@ -0,0 +1,535 @@
+        Initialize the database service.
+
+        Args:
+            supabase_client: Optional Supabase client instance. If not provided,
+                           will use the default client from client_factory.
+            embedding_service: Optional embedding service instance. If not provided,
+                             will create a new instance.
+        """
+        self.supabase_client = supabase_client or get_supabase_client()
+        self.embedding_service = embedding_service or EmbeddingService()
+
+    def add_documents_to_supabase(
```

Pylint (3.3.7)

[refactor] Too many arguments (7/5)

(R0913)